### ISSUE TITLE
Error when installing Release 0.1 on Anki 2.1.49 (Linux)

### ISSUE THREAD

**Issue by itsupera** [Ben-Kerman/anki-jrp#2]
I have tried to install the release 0.1 (IPADIC version) from the .ankiaddon file, but it looks like a library is missing.
Here are the debug info:
```
Anki 2.1.49 (dc80804a) Python 3.8.1 Qt 5.15.1 PyQt 5.15.1
Platform: Linux
Flags: frz=True ao=True sv=2
Add-ons, last update check: 2022-07-20 08:49:15

Caught exception:
Traceback (most recent call last):
  File "aqt/main.py", line 1634, in onAppMsg
  File "aqt/main.py", line 1189, in installAddon
  File "aqt/addons.py", line 1595, in installAddonPackages
  File "aqt/addons.py", line 467, in processPackages
  File "aqt/addons.py", line 402, in install
  File "aqt/addons.py", line 439, in _install
  File "zipfile.py", line 1628, in extract
  File "zipfile.py", line 1698, in _extract_member
  File "zipfile.py", line 1569, in open
  File "zipfile.py", line 817, in __init__
  File "zipfile.py", line 718, in _get_decompressor
  File "zipfile.py", line 695, in _check_compression
RuntimeError: Compression requires the (missing) lzma module
```

I also try using the source code, here is the traceback at launch:
```
⁨Traceback (most recent call last):
  File "aqt/addons.py", line 230, in loadAddons
  File "/home/didier/.local/share/Anki2/addons21/anki-jrp/__init__.py", line 1, in <module>
    from . import ankilib
  File "/home/didier/.local/share/Anki2/addons21/anki-jrp/ankilib/__init__.py", line 1, in <module>
    from . import hooks
  File "/home/didier/.local/share/Anki2/addons21/anki-jrp/ankilib/hooks.py", line 4, in <module>
    from . import browser, editor, global_vars, main_menu, updates
  File "/home/didier/.local/share/Anki2/addons21/anki-jrp/ankilib/browser.py", line 16, in <module>
    from . import global_vars as gv
  File "/home/didier/.local/share/Anki2/addons21/anki-jrp/ankilib/global_vars.py", line 3, in <module>
    from lzma import LZMAError
  File "/home/dae/venv/lib/python3.8/site-packages/PyInstaller-4.0.dev0+g2886519-py3.8.egg/PyInstaller/loader/pyimod03_importers.py", line 625, in exec_module
  File "lzma.py", line 27, in <module>
ModuleNotFoundError: No module named '_lzma'
```
I thought of a conflict with another add-on, but I'm getting the same result with all other add-ons disabled.

**Comment by Ben-Kerman** [comment]
This is probably caused by your distro's Python package not including the full Python standard library (namely it's missing the [LZMA module](https://docs.python.org/3/library/lzma.html))

Try switching to a version that does include the full library, or if that's not possible manually unzip the data files (`data/accents.xz` and `data/variants.xz` in the addon's install dir) and remove usages of `lzma` from `pylib/dictionary.py` and `ankilib/global_vars.py`

These patches should work (although I didn't test them):

```patch
--- a/pylib/dictionary.py
+++ b/pylib/dictionary.py
@@ -1,5 +1,4 @@
 # This project is licensed under the terms of the GNU GPL v3: https://www.gnu.org/licenses/; © 2022 Ben Kerman
-import lzma
 import sys
 from dataclasses import dataclass
 from typing import Dict, Generic, Iterable, List, Optional, Sequence, TextIO, Type, TypeVar
@@ -19,7 +18,7 @@ class BasicDict(Generic[T]):
         self.readings = {}
 
         fd: TextIO
-        with lzma.open(path, "rt", encoding="utf-8") as fd:
+        with open(path, "rt", encoding="utf-8") as fd:
             for line in (rl.rstrip("\r\n") for rl in fd):
                 if line.startswith("#"):
                     continue
```

```patch
--- a/ankilib/global_vars.py
+++ b/ankilib/global_vars.py
@@ -1,6 +1,5 @@
 # This project is licensed under the terms of the GNU GPL v3: https://www.gnu.org/licenses/; © 2022 Ben Kerman
 import os.path
-from lzma import LZMAError
 from os.path import dirname
 from typing import Optional, Type, TypeVar
 
@@ -72,7 +71,7 @@ def load_dict():
         else:
             try:
                 return BasicDict(entry_t, path)
-            except (OSError, LZMAError) as e:
+            except OSError as e:
                 aqt.mw.taskman.run_on_main(
                     lambda: aqt.utils.showWarning(
                         f"Loading {path} failed, reading/accent generation will not work:\n{e}",
@@ -80,8 +79,8 @@ def load_dict():
                     )
                 )
 
-    acc_dic = load_data("accent", "accents.xz", AccentEntry)
-    var_dic = load_data("variants", "variants.xz", VariantEntry)
+    acc_dic = load_data("accent", "accents", AccentEntry)
+    var_dic = load_data("variants", "variants", VariantEntry)
     if acc_dic and var_dic:
         global dictionary
         dictionary = Dictionary(acc_dic, var_dic)

```

**Comment by itsupera** [comment]
Thank you for your help, the patch did work!

FYI I've instigated it a bit more.
TL;DR: My guess is that this is a bug in Anki 2.1.49, specific to the Linux version. Let me know if you did test it on Linux, if so my guess would be wrong.
- From the stacktrace, I understand that a PyInstaller-related module is "shadowing" the standard lib `lzma`. That module is most likely internal to Anki because the path starts with `/home/dae` and `dae` is the username of Anki's lead dev.
- As far as I understand the Python binaries used by Anki are internal to it, called through a PyQt5 binary part of the installation. The Python version indicated by Anki does not match with any globally installed Python version on my system anyway.
- When printing `sys.path` just before the error, it only points to the Anki installation and the addons21 folder, but nothing related to the Python installation of my system.
- Finally, when switching to Anki 2.1.50, the error disappears. However (unsurprisingly) the add-on is not usable due to another error when trying to add or remove (`JS error /_anki/legacyPageData?id=140447298036928:1 Uncaught ReferenceError: getCurrentField is not defined`)

**Comment by Ben-Kerman** [comment]
Thanks for investigating. I'm also on Linux but I don't use the prebuilt binaries, and my Anki package runs on my system's main Python install, so there's really no way this error could've happened to me

Since the issue can be solved by switching to a newer version of Anki I won't try to fix it specifically as it'll go away with next update (which will support whatever the most recent version of Anki is by the time I get to work on it) anyway. I'll leave this issue open until then so anyone else having the same issue can easily find the solution

**Comment by Ben-Kerman** [comment]
Should be fixed with 0.2 on Anki 2.1.55. Please open another issue if not
